### ISSUE TITLE
XSS Vulnerability on closeText option of Dialog jQuery UI

### ISSUE THREAD

**Issue by phatly27** [jquery/api.jqueryui.com#281]
I couldn't submit the bug to the website http://bugs.jqueryui.com/newticket.
## VULNERABILITY DETAILS
A potential bug enables us to inject the XSS content into closeText option using component ui dialog. As original of jQuery UI(https://api.jqueryui.com/dialog/#option-closeText), we shall not accept any HTML string inside it.

## VERSION
Any site using the latest version jQuery UI 1.11.4 .

## REPRODUCTION CASE
* Create a new HTML page.
* Inject this content into new page.

```html
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
    <title>XSS in closeText option of component ui dialog</title>
    
    <script src="https://code.jquery.com/jquery-2.1.4.js"></script>
    <script src="https://code.jquery.com/ui/1.11.4/jquery-ui.js"></script>
	<link rel="stylesheet" type="text/css" href="http://code.jquery.com/ui/1.9.1/themes/base/
jquery-ui.css">
    <script>
        $(document).ready(function () {
			$('#dialog').dialog({ closeText: '<script>alert("XSS")<\/script>' });
        });
    </script>
</head>
<body>
    <div id="dialog" title="Dialog Title">Content here!</div>
</body>
</html>
```
* A alert popup was shown.
* Completed.

## IN CONCLUSION
We expect that the html string isn't allowed in the closeText option as well as the popup alert not shown. If it displays, any attacker can take advantage of injecting the malicious XSS content into website.

Please see details at here  http://jsfiddle.net/0wjdtcc6/

**Comment by testuser100101** [comment]
hello guys 
if you found a page using closeText option on a random website how can you get a working xss ?

**Comment by chrisdclarke** [comment]
just a quick question,...  forgive my ignorance,,   why is it necessary to have a \ before the /script>....

**Comment by energyempath** [comment]
Has this issue been fixed in a later release?

**Comment by anivja** [comment]
Hi, Even after upgrading to latest 1.12.0, I am able to replicate this issue. I used class 'ui-dialog' instead of id as dialog.

**Comment by oze4** [comment]
> just a quick question,... forgive my ignorance,, why is it necessary to have a \ before the /script>....

I believe because `\` is the escape character.. the `\` essentially escapes the next character, which is `/`, therefore, the `/` is rendered and not removed.

**Comment by SaFiSec** [comment]
Hey @1001binary 👋

I was checking out your old bug report on CVE-2016-7103 (XSS via the `closeText` option in jQuery UI Dialog – issue #281).

I wanted to ask: if a site is still using jQuery UI 1.11.4 and doesn't sanitize or encode the `closeText` input, how would a real attacker actually trigger the XSS popup in that context?

Like — do you have a real-world example of this being used on a vulnerable site where attacker-controlled input reflects into the dialog config?

Appreciate the original report, and curious about how it might still apply in live scenarios.

Thanks!
HUNTER
